tools/scxtop/src/symbol_data.rs

```diff
@@ -8,6 +8,191 @@ use blazesym::symbolize::{Input, Sym, Symbolizer};
 use blazesym::Pid;
 use std::collections::{BTreeMap, HashMap};
 use std::path::PathBuf;
+use std::sync::mpsc;
+use std::thread;
+
+const MAX_SYMBOLS: usize = 5000;
+const MAX_STACK_TRACES_PER_SYMBOL: usize = 10; // Limit stack traces per symbol
+const MAX_STACK_DEPTH: usize = 255;
+
+/// Request for symbolization sent to the worker thread
+#[derive(Debug, Clone)]
+pub struct SymbolizationRequest {
+    pub address: u64,
+    pub pid: u32,
+    pub is_kernel: bool,
+    pub request_id: u64,
+}
+
+/// Response from symbolization worker thread
+#[derive(Debug, Clone)]
+pub struct SymbolizationResponse {
+    pub request_id: u64,
+    pub address: u64,
+    pub symbol_info: SymbolInfo,
+}
+
+/// Handle for the async symbolization worker
+pub struct AsyncSymbolizer {
+    request_sender: mpsc::Sender<SymbolizationRequest>,
+    response_receiver: mpsc::Receiver<SymbolizationResponse>,
+    next_request_id: u64,
+}
+
+impl std::fmt::Debug for AsyncSymbolizer {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("AsyncSymbolizer")
+            .field("next_request_id", &self.next_request_id)
+            .finish()
+    }
+}
+
+impl AsyncSymbolizer {
+    pub fn new() -> Self {
+        let (req_tx, req_rx) = mpsc::channel::<SymbolizationRequest>();
+        let (resp_tx, resp_rx) = mpsc::channel::<SymbolizationResponse>();
+
+        // Spawn the worker thread
+        thread::spawn(move || {
```

Comment:
Might be worth switching to `tokio::spawn` and `tokio::mpsc` for uniformity, you can still do a receive in a sync context with `try_recv` which I think is appropriate for this use case.